### PR TITLE
Feature/identity create, edit, delete

### PR DESCRIPTION
**Pull request by AniJan**

### CODE REVIEW

File: identity-api/identity-api-facade/src/main/java/com/sflpro/identity/api/endpoints/IdentityEndpoint.java [hunk 1]
```diff
@@ -85,8 +88,9 @@ public IdentityDto update(@NotNull @PathParam("identityId") final String identit
     @Path("/{identityId}")
     @Deprecated
     @Transactional
-    public IdentityDto delete(@NotNull @PathParam("identityId") final String identityId) {
-        return new IdentityDto();
+    public Response delete(@NotNull @PathParam("identityId") final String identityId) {
```
Comment (sfl-sonar) [inline identity-api/identity-api-facade/src/main/java/com/sflpro/identity/api/endpoints/IdentityEndpoint.java]: MAJOR Add the missing @deprecated Javadoc tag. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AMissingDeprecatedCheck)
INFO Do not forget to remove this deprecated code someday. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1133)

**Comment by sfl-sonar** [comment]
SonarQube analysis reported 2 issues
* ![MAJOR][MAJOR] 1 major
* ![INFO][INFO] 1 info

Watch the comments in this conversation to review them.

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'

File: identity-api/identity-api-facade/src/main/java/com/sflpro/identity/api/endpoints/IdentityEndpoint.java [hunk 2]
```diff
@@ -83,10 +86,10 @@ public IdentityDto update(@NotNull @PathParam("identityId") final String identit
     @ApiOperation("Delete identity")
     @DELETE
     @Path("/{identityId}")
-    @Deprecated
     @Transactional
-    public IdentityDto delete(@NotNull @PathParam("identityId") final String identityId) {
-        return new IdentityDto();
+    public Response delete(@NotNull @PathParam("identityId") final String identityId) {
```
Comment (mikron) [inline identity-api/identity-api-facade/src/main/java/com/sflpro/identity/api/endpoints/IdentityEndpoint.java]: Please use ApiResponseDto so the response will also contain information about time spent on request and version of the application

File: identity-core/identity-core-services-impl/src/main/java/com/sflpro/identity/core/services/identity/IdentityServiceImpl.java [hunk 3]
```diff
@@ -178,4 +188,34 @@ public boolean chkSecretCorrectAndIdentityActive(final Identity identity, final
...
+        final Identity identity = new Identity();
+        identity.setContactMethod(IdentityContactMethod.valueOf(addRequest.getContactMethod()));
+        identity.setSecret(secretHashHelper.hashSecret(addRequest.getSecret()));
+        identity.setDescription(addRequest.getDescription());
+        identity.setStatus(IdentityStatus.ACTIVE);
+        if (!addRequest.getStatus().isEmpty()) {
+            identity.setStatus(IdentityStatus.valueOf(addRequest.getStatus()));
+        }
+        if (!addRequest.getCreatorId().isEmpty()) {
+            identity.setCreatorId(identityRepository.getOne(addRequest.getCreatorId()));
```
Comment (mikron) [inline identity-core/identity-core-services-impl/src/main/java/com/sflpro/identity/core/services/identity/IdentityServiceImpl.java]: We should have some method, which will find not deleted identities

File: identity-core/identity-core-db-repositories/src/main/java/com/sflpro/identity/core/db/repositories/IdentityRepository.java [hunk 4]
```diff
@@ -14,4 +14,6 @@
 public interface IdentityRepository extends JpaRepository<Identity, String> {
 
     Optional<Identity> findByDeletedIsNullAndId(final String id);
+
+    Identity findOneByIdAndDeletedIsNull(final String id);
```
Comment (mikron) [inline identity-core/identity-core-db-repositories/src/main/java/com/sflpro/identity/core/db/repositories/IdentityRepository.java]: Is the upper method doing the same?

File: identity-core/identity-core-services-impl/src/main/java/com/sflpro/identity/core/services/identity/IdentityServiceImpl.java [hunk 5]
```diff
@@ -178,4 +188,34 @@ public boolean chkSecretCorrectAndIdentityActive(final Identity identity, final
...
+        final Identity identity = new Identity();
+        identity.setContactMethod(IdentityContactMethod.valueOf(addRequest.getContactMethod()));
+        identity.setSecret(secretHashHelper.hashSecret(addRequest.getSecret()));
+        identity.setDescription(addRequest.getDescription());
+        identity.setStatus(IdentityStatus.ACTIVE);
+        if (!addRequest.getStatus().isEmpty()) {
+            identity.setStatus(IdentityStatus.valueOf(addRequest.getStatus()));
+        }
+        if (!addRequest.getCreatorId().isEmpty()) {
+            identity.setCreatorId(identityRepository.findByDeletedIsNullAndId(addRequest.getCreatorId()).get());
```
Comment (sfl-sonar) [inline identity-core/identity-core-services-impl/src/main/java/com/sflpro/identity/core/services/identity/IdentityServiceImpl.java]: MAJOR Call "Optional#isPresent()" before accessing the value. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS3655)

File: identity-core/identity-core-services-impl/src/main/java/com/sflpro/identity/core/services/identity/IdentityServiceImpl.java [hunk 6]
```diff
@@ -178,4 +188,36 @@ public boolean chkSecretCorrectAndIdentityActive(final Identity identity, final
...
+        final Identity identity = new Identity();
+        identity.setContactMethod(IdentityContactMethod.valueOf(addRequest.getContactMethod()));
+        identity.setSecret(secretHashHelper.hashSecret(addRequest.getSecret()));
+        identity.setDescription(addRequest.getDescription());
+        identity.setStatus(IdentityStatus.ACTIVE);
+        if (!addRequest.getStatus().isEmpty()) {
+            identity.setStatus(IdentityStatus.valueOf(addRequest.getStatus()));
+        }
+        if (!addRequest.getCreatorId().isEmpty()) {
+            if (identityRepository.findByDeletedIsNullAndId(addRequest.getCreatorId()).isPresent()) {
```
Comment (sfl-sonar) [inline identity-core/identity-core-services-impl/src/main/java/com/sflpro/identity/core/services/identity/IdentityServiceImpl.java]: MAJOR Merge this if statement with the enclosing one. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1066)

File: identity-core/identity-core-services-impl/src/main/java/com/sflpro/identity/core/services/identity/IdentityServiceImpl.java [hunk 7]
```diff
@@ -178,4 +188,36 @@ public boolean chkSecretCorrectAndIdentityActive(final Identity identity, final
...
+                identity.setCreatorId(identityRepository.findByDeletedIsNullAndId(addRequest.getCreatorId()).get());
```
Comment (sfl-sonar) [inline identity-core/identity-core-services-impl/src/main/java/com/sflpro/identity/core/services/identity/IdentityServiceImpl.java]: MAJOR Call "Optional#isPresent()" before accessing the value. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS3655)

**Comment by sfl-sonar** [comment]
SonarQube analysis reported 2 issues
* ![MAJOR][MAJOR] 2 major

Watch the comments in this conversation to review them.

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'

File: identity-core/identity-core-services-impl/src/main/java/com/sflpro/identity/core/services/identity/IdentityServiceImpl.java [hunk 8]
```diff
@@ -178,4 +188,34 @@ public boolean chkSecretCorrectAndIdentityActive(final Identity identity, final
...
+        final Identity identity = new Identity();
+        identity.setContactMethod(IdentityContactMethod.valueOf(addRequest.getContactMethod()));
+        identity.setSecret(secretHashHelper.hashSecret(addRequest.getSecret()));
+        identity.setDescription(addRequest.getDescription());
+        identity.setStatus(IdentityStatus.ACTIVE);
+        if (!addRequest.getStatus().isEmpty()) {
+            identity.setStatus(IdentityStatus.valueOf(addRequest.getStatus()));
+        }
+        if (!addRequest.getCreatorId().isEmpty() && identityRepository.findByDeletedIsNullAndId(addRequest.getCreatorId()).isPresent()) {
+            identity.setCreatorId(identityRepository.findByDeletedIsNullAndId(addRequest.getCreatorId()).get());
```
Comment (sfl-sonar) [inline identity-core/identity-core-services-impl/src/main/java/com/sflpro/identity/core/services/identity/IdentityServiceImpl.java]: MAJOR Call "Optional#isPresent()" before accessing the value. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS3655)

**Comment by sfl-sonar** [comment]
SonarQube analysis reported 1 issue
* ![MAJOR][MAJOR] 1 major

Watch the comments in this conversation to review them.

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'

**Comment by dnavre** [comment]
This resolved #17

**Pull request merged**